khal/khalendar/event.py

```diff
@@ -405,7 +405,13 @@ def categories(self):
         return self._vevents[self.ref].get('CATEGORIES', '')
 
     def update_categories(self, categories):
-        self._vevents[self.ref]['CATEGORIES'] = categories
+        if not categories.strip():
+            self._vevents[self.ref]['CATEGORIES'] = categories
+        else:
+            try:
+                del self._vevents[self.ref]['CATEGORIES']
```

Comment:
I think it would be better to do
```python
try:
    self._vevents[self.ref].pop('CATEGORIES')
except KeyError:
    pass
```
or
```python
self._events[self.ref].pop('CATEGORIES', False)
```